Change needs to be added to FormlyTemplateOptions

**I'm submitting a ...**  (check one with "x")
```
[X] bug report
[ ] feature request
[ ] support request
```

**Current behavior**
The code is as follows...

```
export interface FormlyFieldConfig {
   ...
   change?: Function;
   ...
}
```

however, per http://plnkr.co/edit/OfcprFL35FxG3cHpF0Hz?p=preview, if I try to use this property it doesn't work like this `this.connectorFields[0].change`. I instead need to add it to the FormlyTemplateOptions like this... `this.connectorFields[0].templateOptions.change`.

Now in plunker this works OK but if you are transpiling with something like webpack you get...

`(27,49): error TS2339: Property 'change' does not exist on type 'FormlyTemplateOptions'`

so I instead have to use `this.connectorFields[0].templateOptions["change"]`

**Expected behavior**
I would like to be able to use `this.connectorFields[0].templateOptions.change` instead

**Minimal reproduction of the problem with instructions**

See plunker

--- a/demo/hello.ts
+++ b/demo/hello.ts
@@ -302,12 +302,12 @@ export class HelloApp {
           },
           updateOn: 'default blur',
         },
+        focus: true,
         templateOptions: {
           rows: 5,
           cols: 20,
           label: 'Message (Type a paragraph...)',
           description: 'Please enter atleast 150 characters',
-          focus: true,
         },
       }, {
         key: 'toggleVal',

--- a/src/core/components/formly.attributes.spec.ts
+++ b/src/core/components/formly.attributes.spec.ts
@@ -32,7 +32,7 @@ describe('FormlyAttributes Component', () => {
         expect(elm.getAttribute('placeholder')).toBe('Title');
         expect(elm.getAttribute('tabindex')).toBe('5');
         expect(elm.getAttribute('step')).toBe('2');
-        expect(fixture.componentInstance.field.templateOptions.focus).toBeFalsy();
+        expect(fixture.componentInstance.field.focus).toBeFalsy();
       });
 
       it('should change element attribute on edit templateOptions', () => {
@@ -41,9 +41,9 @@ describe('FormlyAttributes Component', () => {
 
         fixture.componentInstance.field = {
           key: 'title',
+          focus: true,
           templateOptions: {
             placeholder: 'Title Edit',
-            focus: true,
           },
         };
 
@@ -52,7 +52,7 @@ describe('FormlyAttributes Component', () => {
         expect(elm.getAttribute('placeholder')).toBe('Title Edit');
         expect(elm.getAttribute('tabindex')).toBe(null);
         expect(elm.getAttribute('step')).toBe(null);
-        expect(fixture.componentInstance.field.templateOptions.focus).toBeTruthy();
+        expect(fixture.componentInstance.field.focus).toBeTruthy();
       });
   });
 
@@ -61,23 +61,23 @@ describe('FormlyAttributes Component', () => {
       const fixture = createTestComponent('<input type="text" [formlyAttributes]="field">');
       const elm = getFormlyAttributesElement(fixture.nativeElement);
 
-      fixture.componentInstance.field = { templateOptions: { focus: true } };
+      fixture.componentInstance.field = { focus: true, templateOptions: {} };
       fixture.detectChanges();
       expect(document.activeElement === elm).toBeTruthy();
 
-      fixture.componentInstance.field = { templateOptions: { focus: false } };
+      fixture.componentInstance.field = { focus: false, templateOptions: {} };
       fixture.detectChanges();
       expect(document.activeElement === elm).toBeFalsy();
     });
 
-    it('should change templateOptions focus when the element is focused', () => {
+    it('should change field focus when the element is focused', () => {
       const fixture = createTestComponent('<input type="text" [formlyAttributes]="field">');
       const directive = fixture.debugElement.query(By.directive(FormlyAttributes));
 
       directive.triggerEventHandler('focus', {});
       fixture.detectChanges();
 
-      expect(fixture.componentInstance.field.templateOptions.focus).toBeTruthy();
+      expect(fixture.componentInstance.field.focus).toBeTruthy();
     });
   });
 });
@@ -85,12 +85,12 @@ describe('FormlyAttributes Component', () => {
 @Component({selector: 'formly-formly-attributes-test', template: '', entryComponents: []})
 class TestComponent {
   field: FormlyFieldConfig = {
+    focus: false,
     key: 'title',
     templateOptions: {
       placeholder: 'Title',
       tabindex: 5,
       step: 2,
-      focus: false,
     },
   };
 }

--- a/src/core/components/formly.attributes.ts
+++ b/src/core/components/formly.attributes.ts
@@ -26,7 +26,7 @@ export class FormlyAttributes implements OnInit, OnChanges {
 
   ngOnInit() {
     this.focusDispatcher.listen((key: String) =>
-      this.field.templateOptions.focus = this.field.key === key);
+      this.field.focus = this.field.key === key);
   }
 
   ngOnChanges(changes: SimpleChanges) {
@@ -54,9 +54,9 @@ export class FormlyAttributes implements OnInit, OnChanges {
           }
         });
 
-      if (templateOptions.focus || (previousOptions.focus !== undefined && previousOptions.focus !== templateOptions.focus)) {
-        this.renderer.invokeElementMethod(this.elementRef.nativeElement, templateOptions.focus ? 'focus' : 'blur', []);
-        if (templateOptions.focus) {
+      if (this.field.focus || (changes['field'].previousValue.focus !== undefined && changes['field'].previousValue.focus !== this.field.focus)) {
+        this.renderer.invokeElementMethod(this.elementRef.nativeElement, this.field.focus ? 'focus' : 'blur', []);
+        if (this.field.focus) {
           // TODO: Raise a Event which can be used for streaming
           this.focusDispatcher.notify(this.field.key);
         }

--- a/src/core/components/formly.field.config.ts
+++ b/src/core/components/formly.field.config.ts
@@ -15,13 +15,7 @@ export interface FormlyFieldConfig {
   className?: string;
   type?: string;
   expressionProperties?: any;
-  focus?: boolean | Function;
-  blur?: Function;
-  keyup?: Function;
-  keydown?: Function;
-  click?: Function;
-  change?: Function;
-  keypress?: Function;
+  focus?: boolean;
   modelOptions?: any;
   lifecycle?: FormlyLifeCycleOptions;
   defaultValue?: any;
@@ -37,7 +31,6 @@ export interface FormlyTemplateOptions {
   rows?: number;
   cols?: number;
   description?: string;
-  focus?: boolean;
   hidden?: boolean;
   max?: number;
   min?: number;
@@ -47,6 +40,13 @@ export interface FormlyTemplateOptions {
   required?: Boolean;
   tabindex?: number;
   step?: number;
+  focus?: Function;
+  blur?: Function;
+  keyup?: Function;
+  keydown?: Function;
+  click?: Function;
+  change?: Function;
+  keypress?: Function;
   [additionalProperties: string]: any;
 }
 

